### PR TITLE
Convolution module

### PR DESCRIPTION
**Pull request by mabevillar**
- Creation of V1.0 of the convolution module
- Addition of 3 functions in utils.py which are required for running the convolution

### CODE REVIEW

**Comment by raoanirudh** [comment]
A couple of suggestions: 
1. For now, we're using the repository [rmtk_data](https://github.com/GEMScienceTools/rmtk_data) to store all the input and output files produced by the rmtk scripts and notebooks. Could you transfer all of the .csv and .gmr files to that repository and make sure that the .out files produced by OpenSees are also written to that repository?
2. The file [~$Hazard.xlsx](https://github.com/GEMScienceTools/rmtk/pull/38/files#diff-3c6741eb974337cebfb4da0b299814f4) is a temporary file; it should be deleted.

File: rmtk/vulnerability/common/utils.py [hunk 1]
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
...
+       for ids in range(len(cent_value)):
+          mean = cent_value[ids]
+          stdev = line[ids]
+          log_mean.append(mean)
+          log_stdev.append(stdev)
+       
+    #Store all the data in the dictionary
+    fragility_model =  {'Damage_states':None,'log_mean':None,'log_stdev':None}
+    fragility_model['Damage_states'] = damage_states
+    fragility_model['log_mean'] = log_mean
```
Comment (raoanirudh) [inline rmtk/vulnerability/common/utils.py]: Please check if this is consistent with the other definitions, which use `fragility_model['logmean']`

File: rmtk/vulnerability/common/utils.py [hunk 2]
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
...
+    prob_exceedance = []
+    IM_type = []
+    for line in data:
+       if line[0] == 'PGA':
+          IM_type.append(line[0])
+          for value in line[1:]:
+             if isNumber(value):    
+                IMLs.append(float(value))
+      if line[0] == 'Sa':
+         IM_type.append(line[0])
```
Comment (raoanirudh) [inline rmtk/vulnerability/common/utils.py]: Please check the indentation for all lines in this function.

File: rmtk/vulnerability/common/utils.py [hunk 3]
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
...
+      hazard_curve['IMLs'] = IMLs
+      hazard_curve['PoE'] = prob_exceedance
+      hazard_curve['IM_Type'] = IM_type
+      
+      return hazard_curve
+         
+def plot_hazard_curve(hazard_curve):
+#This function plots the hazard curve
+
+   IMLs = hazard_curve['IMLs']
```
Comment (raoanirudh) [inline rmtk/vulnerability/common/utils.py]: Please check the indentation for all lines in this function.

File: rmtk/vulnerability/common/utils.py [hunk 4]
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
...
+   PoE = hazard_curve['PoE']
+   plt.plot(IMLs,PoE,color='g',linewidth=2)
+   plt.xlabel(hazard_curve['IM_Type'][0] + ' [g]',fontsize = 10)
+   plt.ylabel('Probability of Exceedance',fontsize = 10)
+
+def read_frag_model(input_file):    
+#This function reads a fragility model and stores it in a dictionary
+
+    damage_states = []
```
Comment (raoanirudh) [inline rmtk/vulnerability/common/utils.py:1771]: Please see above comment about indentation.

**Comment by raoanirudh** [comment]
@mabevillar Is the file [Frag_model.png](https://github.com/mabevillar/rmtk/blob/master/rmtk/vulnerability/derivation_fragility/hybrid_methods/CSM/OutFiles/Frag_model.png) required for the CSM module, or is it just an output file that can be removed?

File: rmtk/vulnerability/Convolution_module/Convolution.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+import csv
+import matplotlib.pyplot as plt
+from rmtk.vulnerability.common import utils
+from scipy.stats import lognorm
+
+def calculate_convolution(hazard_curve,fragility_model,return_period,damage_state,save,plot,typology):
+#This function calculates the annual probability of collapse given a hazard 
+#curve and a fragility model
+
+   rate_exceed = []
```
Comment (raoanirudh) [inline rmtk/vulnerability/Convolution_module/Convolution.py]: Please check the indentation for all functions in this module ([PEP 8 link for reference](https://www.python.org/dev/peps/pep-0008/#indentation)).

**Comment by mabevillar** [comment]
@raoanirudh : The Frag_model.png in the CSM is just an output file. You can delete it.

**Comment by raoanirudh** [comment]
Looks good.

**Pull request merged**